platform/viewer/src/routes/Mode/Mode.tsx

```diff
@@ -294,28 +305,26 @@ export default function ModeRoute({
         ) ?? {};
 
       if (route.init) {
-        return await route.init(
-          {
-            servicesManager,
-            extensionManager,
-            hotkeysManager,
-            studyInstanceUIDs,
-            dataSource,
-            filters,
-          },
-          hangingProtocolIdToUse
-        );
-      }
-
-      return defaultRouteInit(
-        {
+        return await route.init({
           servicesManager,
+          extensionManager,
+          hotkeysManager,
           studyInstanceUIDs,
           dataSource,
           filters,
-        },
-        hangingProtocolIdToUse
-      );
+          hangingProtocol: hangingProtocolIdToUse,
+          hangingProtocolStageIndex: hangingProtocolStageIndexToUse,
+        });
+      }
+
+      return defaultRouteInit({
+        servicesManager,
+        studyInstanceUIDs,
+        dataSource,
+        filters,
+        hangingProtocol: hangingProtocolIdToUse,
```

Comment:
This needs to start getting passed in as a set of options to the route init, as there may well be others that we want here.